src/main/java/DateValidation.java

```diff
@@ -0,0 +1,38 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Handles the validation of dates for Deadlines.
+ */
+public class DateValidation {
+
+    /**
+     * Checks if the input is in the correct format of "yyyy-mm-dd".
+     * @param date Input to be checked.
+     * @return A LocalDate object created from the input.
+     * @throws DukeException On invalid input. Input format is wrong.
+     */
+    public static LocalDate handleDate(String date) throws DukeException {
+        LocalDate verifiedDate = null;
+        try {
+            verifiedDate = LocalDate.parse(date);
+            return verifiedDate;
+        } catch (DateTimeParseException e) {
+            throw new DukeException(":( Date format is invalid! Please enter in yyyy-mm-dd format!");
+        }
+    }
+
+    /**
+     * Creates a LocalDate object from a string.
+     * @param date String that is used to create a LocalDate object.
+     * @return A LocalDate object created.
+     */
+    public LocalDate convertDate(String date) {
+        DateTimeFormatter format = DateTimeFormatter.ofPattern("MMM dd yyyy");
```

Comment:
"Avoid magic numbers" https://nus-cs2103-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers

Might want to use a named constant for this?